apps/test/unit/weblab/WebLabTest.js

```diff
@@ -81,6 +88,176 @@ describe('WebLab', () => {
+      try {
+        await config.afterClearPuzzle();
+      } catch (error) {
+        didCatch = true;
+        expect(error.message).to.equal(
+          'deleteAll succeeded, weblab handling reload to avoid saving'
+        );
+      }
+      expect(didCatch).to.equal(true);
+      expect(weblab.fileEntries).to.equal(null);
+      filesApi.deleteAll.restore();
+    });
+
+    it('throws error showing deleteAll failed when filesApi deleteAll fails', async () => {
+      sinon.stub(filesApi, 'deleteAll').callsFake((success, error) => {
+        error({status: 'status'});
+      });
+      sinon.stub(console, 'warn');
+      weblab.fileEntries = 'entries';
+      let didCatch = false;
+      try {
+        await config.afterClearPuzzle();
+      } catch (error) {
+        expect(console.warn).to.have.been.calledOnceWith(
+          'WebLab: error deleteAll failed: status'
+        );
+        didCatch = true;
+        expect(error.message).to.equal('status');
+      }
+      expect(didCatch).to.equal(true);
+      expect(weblab.fileEntries).to.equal('entries');
+      filesApi.deleteAll.restore();
+      console.warn.restore();
+    });
+  });
+
+  describe('onToggleInspector', () => {
+    let brambleHost;
+    beforeEach(() => {
+      brambleHost = {
+        disableInspector: sinon.stub(),
+        enableInspector: sinon.stub()
+      };
+      weblab.brambleHost = brambleHost;
+      stubRedux();
+    });
+
+    afterEach(() => {
+      restoreRedux();
+    });
+
+    it('disables inspector if inspectorOn', () => {
+      sinon.stub(getStore(), 'getState').returns({inspectorOn: true});
+      weblab.onToggleInspector();
+      expect(brambleHost.disableInspector).to.have.been.calledOnce;
+    });
+
+    it('enables inspector if inspectorOn false', () => {
+      sinon.stub(getStore(), 'getState').returns({inspectorOn: false});
+      weblab.onToggleInspector();
+      expect(brambleHost.enableInspector).to.have.been.calledOnce;
+    });
+  });
+
+  describe('onMount', () => {
+    let config;
+    beforeEach(() => {
+      stubStudioApp();
+      stubRedux();
+      config = {
+        containerId: 'container-id'
+      };
+      weblab.studioApp_ = studioApp();
+      sinon.stub(studioApp(), 'setConfigValues_');
+      sinon.stub(studioApp(), 'initProjectTemplateWorkspaceIconCallout');
+      sinon.stub(studioApp(), 'alertIfCompletedWhilePairing');
+      sinon.stub(studioApp(), 'initVersionHistoryUI');
+      sinon.stub(studioApp(), 'initTimeSpent');
+      weblab.level = {unsubmitUrl: 'url'};
+      sinon.stub(dom, 'addClickTouchEvent');
+    });
+
+    afterEach(() => {
+      restoreStudioApp();
+      restoreRedux();
+      dom.addClickTouchEvent.restore();
+    });
+
+    it('adds clickTouchEvent 3 times if there is a finishButton', () => {
+      sinon.stub(document, 'getElementById').returns({className: 'test'});
+      weblab.onMount(config);
+      expect(dom.addClickTouchEvent).to.have.been.calledThrice;
```

Comment:
why does this get called 3 times? i only see 1 call within `onMount`

Reply:
It gets called twice in initializeSubmitHelper and I wasn't sure how to work around that

Reply:
ahh, okay. we can't stub `initializeSubmitHelper` because it doesn't belong to an object, so i think we could clarify the expectations here and in the next test to instead check whether or not `dom.addClickTouchEvent` was called with our `finishButton`: https://www.chaijs.com/plugins/sinon-chai/